### PR TITLE
Clean up domain views

### PR DESCRIPTION
**Pull request by esoergel**
I'm pulling these out of my other branch so they can be merged independently.  (After the migration)
This is based on the branch in https://github.com/dimagi/commcare-hq/pull/9137, so most of the commits come from there.
@dannyroberts @millerdev cc: @nickpell

### CODE REVIEW

**Comment by esoergel** [comment]
(to be merged after the migration)

File: corehq/couchapps/by_domain_doc_type/views/view/map.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function (doc) {
```
Comment (dimagimon) [inline corehq/couchapps/by_domain_doc_type/views/view/map.js]: Missing name in function declaration.

File: corehq/couchapps/by_domain_doc_type_date/views/view/map.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+function (doc) {
```
Comment (dimagimon) [inline corehq/couchapps/by_domain_doc_type_date/views/view/map.js]: Missing name in function declaration.

**Comment by dannyroberts** [comment]
Just rebased onto master. This is good cleanup and we can now easily merge at anytime with basically no reindex now that this is in a separate db.

**Comment by esoergel** [comment]
excellent

**Comment by czue** [comment]
:racehorse: :horse_racing: :racehorse: :horse_racing:

File: corehq/apps/domain/_design/views/docs/map.js [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-    // !code util/doc_utils.js
-
-    var i,
-        date,
-        domains;
-    if (doc.doc_type) {
-        date = get_date(doc);
-        domains = get_domains(doc);
-        for (i = 0; i < domains.length; i += 1) {
-            emit([domains[i], doc.doc_type, date], null);
```
Comment (sravfeyn) [inline corehq/apps/domain/_design/views/docs/map.js]: Hey @esoergel, it looks like [`copy_domain`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domainsync/management/commands/copy_domain.py#L173) uses this view heavily. Isn't this the best method for copy_domain? Should we get this back?

Comment (esoergel) [inline corehq/apps/domain/_design/views/docs/map.js]: `"domain/docs"` was replaced by  `"by_domain_doc_type_date/view"` in the PR this one is based off of: https://github.com/dimagi/commcare-hq/pull/9137/files (You can find some examples in there of how to do the switch - in most cases it's trivial).  I'll copy/paste this to the ticket you linked.

**Pull request merged**